build_rsas_from_etas.gs

```diff
@@ -0,0 +1,390 @@
+    }
+     
+    groupedAssets[adGroupId] = groupedAssets[adGroupId] || {
+      'ad_group_id': adGroupId,
+      'ad_group_name': adGroupName,
+      'campaign_id': campaignId,
+      'campaign_name': campaignName,
+      'headlines': [],
+      'descriptions': [],
+      'path1s': [],
+      'path2s': [],
+      'final_urls': [],
+      'final_mobile_urls': []
+     };
+    
+    pushElementsIfNeeded(groupedAssets[adGroupId]['headlines'], [headline1, headline2, headline3]);
+    pushElementsIfNeeded(groupedAssets[adGroupId]['descriptions'], [description1, description2]);
+    pushElementsIfNeeded(groupedAssets[adGroupId]['path1s'], [path1]);
+    pushElementsIfNeeded(groupedAssets[adGroupId]['path2s'], [path2]);
+    pushElementsIfNeeded(groupedAssets[adGroupId]['final_urls'], [finalUrl]);
+    pushElementsIfNeeded(groupedAssets[adGroupId]['final_mobile_urls'], [finalMobileUrl]);
+    
+  }
+  
+  return groupedAssets;
+}
+
+// Write headers and body to sheet
+function writeToSheet(sheet, groupedAssets, maxNoOfAssets, accountName, accountId) {
+  setHeaders(sheet, maxNoOfAssets);
+  pushDataToSheet(sheet, groupedAssets, maxNoOfAssets, accountName, accountId);
+}
+
+// Set headers in a sheet
+function setHeaders(sheet, maxNoAssets) {
+  
+  // Attribute headers
+  var headers = ["Account Name", "Account ID", "Campaign Name", "Campaign ID", "Ad Group Name", "Ad Group ID"];
+  var numColumns = headers.length;
+  
+  // Asset headers (of variable length)
+  for (var i = 0; i < maxNoAssets['max_headlines']; i++) {
+    headers.push("Headline " + (i+1));
+  }
+  numColumns += maxNoAssets['max_headlines'];
+  
+  for (var i = 0; i < maxNoAssets['max_descriptions']; i++) {
+    headers.push("Description " + (i+1));
+  }
+  numColumns += maxNoAssets['max_descriptions'];
+
+  headers.push("Path 1(s)");
+  for (var i = 1; i < maxNoAssets['max_path1s']; i++) {
+    headers.push("...");
+  }
+  numColumns += Math.max(1,maxNoAssets['max_path1s'], 1);
+  
+  headers.push("Path 2(s)");
+  for (var i = 1; i < maxNoAssets['max_path2s']; i++) {
+    headers.push("...");
+  }
+  numColumns += Math.max(1,maxNoAssets['max_path2s'], 1);
+ 
+  headers.push("Final URL(s)");
+  for (var i = 1; i < maxNoAssets['max_urls']; i++) {
+    headers.push("...");
+  }
+  numColumns += Math.max(1, maxNoAssets['max_urls']);
+  
+  headers.push("Final Mobile URL(s)");
+  for (var i = 1; i < maxNoAssets['max_mobile_urls']; i++) {
+    headers.push("...");
+  }
+  numColumns += Math.max(1, maxNoAssets['max_mobile_urls']);
+  
+  // Font weight row to allow us to set the headers to bold
+  var boldings = [];
+  for (var i = 0; i < numColumns; i++) {
+    boldings.push("bold");
+  }
+  
+  // Set values and bolding
+  sheet.getRange(1, 1, 1, numColumns).setValues([headers]).setFontWeights([boldings]);
+  
+  return null;
+}
+
+// Set outputs for sheet
+function pushDataToSheet(sheet, groupedAssets, maxNoAssets, accountName, accountId) {
```

Comment:
This is nice! 
You might want to try next time to write a modular function that can be reused no matter what the structure of the table is. ie passing a list of dicts and just creates the table no matter the col names etc..

Reply:
Ah nice, that's a good point!

Reply:
Same as above, if you can add to asana for a future version